AsyncDisplayKit/ASContextTransitioning.h

```diff
@@ -0,0 +1,33 @@
+//
+//  ASContextTransitioning.h
+//  AsyncDisplayKit
+//
+//  Created by Levi McCallum on 2/4/16.
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+
+@protocol ASContextTransitioning <NSObject>
+
+/**
+ @abstract The frame for the given node before the transition began.
+ @discussion Returns CGRectNull if the node was not in the hierarchy before the transition.
+ */
+- (CGRect)initialFrameForNode:(ASDisplayNode *)node;
```

Comment:
Frame might be worth including in the API as a convenience using many cases, but for this to be fully generalized, we would have to support setting balance and position along with transform since Frame is not defined with a non identity transform